linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -1190,7 +1219,15 @@ private[emitter] object CoreJSLib {
 
       condDefs(esVersion < ESVersion.ES2015)(
         defineFunction5(VarField.systemArraycopy) { (src, srcPos, dest, destPos, length) =>
-          genCallHelper(VarField.arraycopyGeneric, src.u, srcPos, dest.u, destPos, length)
+          if (useBigIntForLongs) {
+            genCallHelper(VarField.arraycopyGeneric, src.u, srcPos, dest.u, destPos, length)
+          } else {
+            If((src DOT classData) === genClassDataOf(ArrayTypeRef(LongRef, 1)), {
+              genCallHelper(VarField.arraycopyGeneric, src.u, srcPos << 1, dest.u, destPos << 1, length << 1)
```

Comment:
Just to check my understanding: We need not worry about overflow here, because we at most have 32 bits input, but have 52 at our disposal (and use 33 at most).